eam-light-backendweb/src/main/java/ch/cern/cmms/eamlightweb/user/ScreenService.java

```diff
@@ -56,21 +57,38 @@ private void init() {
                                             inforClient.getGridsService().executeQuery(context, gridRequestLayout));
 
         // Populate the parent screen for fetched screens
-        screens.values().stream().forEach(screen -> screen.setParentScreen(functions.get(screen.getScreenCode())));
+        screens.values().stream().forEach(screen -> {
+            screen.setParentScreen(functions.get(screen.getScreenCode()).getParentScreenCode());
+            screen.setStartupAction(functions.get(screen.getScreenCode()).getStartUpModeDisplayCode());
+        });
 
         screenCache.put(userGroup, screens);
         return screens;
     }
 
-    public Map<String, String> getFunctions(InforContext context) throws InforException {
+    public Map<String, EamFunction> getFunctions(InforContext context) throws InforException {
         GridRequest gridRequestLayout = new GridRequest("BSFUNC", 1000);
-        gridRequestLayout.addFilter("parentscreencode", String.join(",", screens), "IN");
+        gridRequestLayout.addFilter("parentscreencode", String.join(",", screens), "IN", JOINER.OR);
+        gridRequestLayout.addFilter("screencode", String.join(",", screens), "IN");
 
-        Map<String, String> functions =  inforClient.getTools().getGridTools().convertGridResultToMap("screencode",
-                "parentscreencode",
+        Map<String, EamFunction> functions =  inforClient.getTools().getGridTools().convertGridResultToMap(EamFunction.class,
+                "screencode",
+                null,
                 inforClient.getGridsService().executeQuery(context, gridRequestLayout));
 
-        screens.forEach(screen -> functions.put(screen, screen));
+        GridRequest startUpActionTypesGridRequest = new GridRequest("BSUCOD_HDR", 300);
+        startUpActionTypesGridRequest.addParam("param.entitycode", "FAQU");
+
+        Map<String, String> startUpActionDescriptionToCode = inforClient.getTools().getGridTools().convertGridResultToMap(
+                "usercodedescription",
+                "systemcode",
+                inforClient.getGridsService().executeQuery(context, startUpActionTypesGridRequest));
+
+        screens.forEach(screen -> functions.computeIfPresent(screen, (key, value) -> {
```

Comment:
Nitpick: I would change `key` and `value` here to `screenCode` and `eamFunction`, to better convey the semantics of key and value.